### PR TITLE
feat: integrate llm-ls

### PR DESCRIPTION
**Pull request by McPatate**

### CODE REVIEW

File: src/bootstrap.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// import * as vscode from "vscode";
```
Comment (mishig25) [inline src/bootstrap.ts:1]: I guess this file is not needed?

File: src/extension.ts [hunk 2]
```diff
@@ -1,76 +1,164 @@
...
+			}
+		}
+	};
 
-  return Promise.resolve();
-}
+	const clientOptions: LanguageClientOptions = {
+		documentSelector: [{ scheme: "file" }],
+	};
+	client = new LanguageClient(
```
Comment (mishig25) [inline src/extension.ts:49]: 🔥

File: src/extension.ts [hunk 3]
```diff
@@ -1,76 +1,164 @@
...
+					temperature: config.get("temperature") as number,
+					do_sample: true,
+					top_p: 0.95,
+				},
+				fim: config.get("fillInTheMiddle") as number,
+				context_window: config.get("contextWindow") as number,
+				tls_skip_verify_insecure: config.get("tlsSkipVerifyInsecure") as boolean,
+			};
+			const completion: Completion[] = await client.sendRequest("llm-ls/getCompletions", params, token);
+			const insertText = completion[0].generated_text;
```
Comment (mishig25) [inline src/extension.ts:124]: what happens (i.e. what does the user see) when `client.sendRequest` errs ?

**Review by mishig25 — APPROVED** [review]
lgtm! Very nice and clean

A small nit (probably for a serperate PR) is that the previous version has a spining status bar which gives a nice UI feedback to the user when the inference is happening

image

File: src/extension.ts (hunk 3 above)
Comment (McPatate) [reply src/extension.ts:124]: Nothing for the user, it's logged somewhere though

File: src/bootstrap.ts (hunk 1 above)
Comment (McPatate) [reply src/bootstrap.ts:1]: Removed it

**Pull request merged**